decorator.rb

```diff
@@ -0,0 +1,11 @@
+require_relative 'nameable'
+
+class Decorator
+  def initialize(nameable)
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
```

Comment:
- [ ] Please make sure the `nameable` variable is readable and writable for the associations to work as expected.
  - You can consider the code `attr_accessor :nameable`.